components/camel-azure/src/main/java/org/apache/camel/component/azure/blob/BlobServiceProducer.java

```diff
@@ -495,4 +500,12 @@ private void closeInputStreamIfNeeded(InputStream inputStream) throws IOExceptio
         }
     }
 
+
+    private void overrideBlobName(Exchange exchange) {
+        String blobName = exchange.getIn().getHeader(OVERRIDE_BLOB_NAME, String.class);
```

Comment:
Ah this is wrong, you should not change the endpoint value. Instead you should return the blob name to use, eg pick from header if exists, if not then fallback and use configured value from endpoint configuratoon

Reply:
@davsclaus Can you please give me a little more guidance regarding what you want. Do you want me to provide the overriden blobName through the getBlobName() method which is in BlobServiceEndpoint class?

Reply:
if exchange header has blob name then 
  to use blob name = exchange header 
otherwise 
  to use blob name = endpoint blob name

And then use the "to use blob name" in the producer